### ISSUE TITLE
run collection FS with 2 meteor app

### ISSUE THREAD

**Issue by particle4dev** [CollectionFS/Meteor-CollectionFS#651]
Hello,
I have a issue when i try to insert image via URL.Basically i have two meteor applications use the same Mongo(mongohq): one for mobile app(1) and one for process data(2). Both of it can upload images. The problem is when app (2) try to upload image, it's crash app (1) and fail to upload.
Here logs on app(1):
```
W20150412-15:44:41.784(7)? (STDERR) 
W20150412-15:44:41.785(7)? (STDERR) stream.js:94
W20150412-15:44:41.785(7)? (STDERR)       throw er; // Unhandled stream error in pipe.
W20150412-15:44:41.786(7)? (STDERR)             ^
W20150412-15:44:41.786(7)? (STDERR) Error: ENOENT, open '<path meteor app>/.meteor/local/cfs/files/_tempstore/images-fWc94zduSt34iKTQx-0.chunk'
```
Any ideas ? Thanks!

**Comment by rhyslbw** [comment]
Duplicate of https://github.com/CollectionFS/Meteor-CollectionFS/issues/635

Follow this PR https://github.com/CollectionFS/Meteor-CollectionFS/pull/667

**Comment by particle4dev** [comment]
@rhyslbw Here my solution, it maybe help someone:
```
FS.debug = true;

var imageStore = new FS.Store.S3("images", {
   ...
});

ImagesCollection = new FS.Collection("images_alias", {
    stores: [imageStore],
    filter: {
        allow: {
            contentTypes: ['image/*'] //allow only images in this FS.Collection
        }
    }
});

ImagesOriginCollection = new FS.Collection("images", {
    stores: [imageStore],
    filter: {
        allow: {
            contentTypes: ['image/*'] //allow only images in this FS.Collection
        }
    }
});

ImagesCollection.on('stored', function(fileObj){
    Fiber(function() {
        var r = ImagesCollection.files.findOne(fileObj._id, {
            transform: function(doc){
                return doc;
            }
        });
        ImagesOriginCollection.files.insert(r);
    }).run();
});
```

**Comment by rhyslbw** [comment]
Thanks for sharing @particle4dev . @bitomule This might be of use as a workaround

**Comment by bitomule** [comment]
@particle4dev and how does this fixes the problem? You don't share ImagesOriginCollection between servers?

Thanks!

**Comment by particle4dev** [comment]
@bitomule I think the problem because CollectionFS use [collection.files](https://github.com/CollectionFS/Meteor-CollectionFS/blob/master/packages/collection/common.js#L112) to tell [FileWorker](https://github.com/CollectionFS/Meteor-CollectionFS/blob/783c9d37de38bead77d73f3701c7ed4adaee7f9d/packages/worker/fileWorker.js#L20) where the uploaded file is stored. When you upload file from app(2), the file doesn't exist on app(1) but FileWorker is still run because they use the same Mongo.
Solution:
App(2)
+ ImagesOriginCollection
+ ImagesCollection ( won't affect to app(1) )

App(1)
+ ImagesOriginCollection

And when the file is uploaded on ImagesCollection, i will copy it to ImagesOriginCollection. Hope that help.

**Comment by bitomule** [comment]
@particle4dev so, on App(2) I always use ImagesCollection on publications and inserts no? App(1) only uses ImagesOriginCollection.

**Comment by particle4dev** [comment]
@bitomule use ImagesCollection on insert, still use ImagesOriginCollection on publications. Try it yourself and you will find out what best for you :)

**Comment by bitomule** [comment]
@particle4dev thanks!

**Comment by bitomule** [comment]
@particle4dev tried and doesn't work for me. I use transformWrite and it seems to be working bad with this fix. Also, this won't work if you have multiple instances of App(2) for example and it will create duplicated info in database.

Thank you for your help @particle4dev but I hope we can have a real fix to this issue ;)

**Comment by particle4dev** [comment]
@bitomule It work in my case because my app(2) is crawler job. Do you think that you should create a "image" service. It work like a restful service. You send a POST request with data when you want to upload the images and You send a GET request, it will be return a url. Just a idea because i think your app is big. :)

**Comment by bitomule** [comment]
@particle4dev is already some kind of api service, as file is uploaded from iOS app but I need collectionFS in App(1) to get upload progress and client side helpers. I just hope great minds in working on this project can find a way to fix this before we start our iOS app beta :)
